torch/_inductor/codegen/triton.py

```diff
@@ -1407,6 +1412,16 @@ def reduction(self, dtype, src_dtype, reduction_type, value):
         reduction_sizes = ["None" for _ in self.range_trees]
         reduction_sizes[-1] = ":"
 
+        # The shape of the reduction dimension changes the output, so
+        # explicitly broadcast the input to the expected shape.
```

Comment:
Added an example that should be clearer. Before this change
```python
tmp0 = ops.constant(1, torch.int64)
tmp1 = ops.reduction(torch.int64, torch.int64, "sum", tmp0)
```
becomes
```python
tmp0 = tl.full([1, 1], 1, tl.int64)
tmp1 = tl.sum(tmp0, 1)[:, None]
```
which is wrong because summing over 1 element is not equivalent to summing over `RBLOCK` elements.